### PR TITLE
Add integration tests for string stdlib funcs

### PR DESCRIPTION
**Pull request by euank**
This adds happy-path coverage for the string stdlib functions.

A couple cases were left out due to bugs discovered over here: https://github.com/tweag/nickel/issues/1003

Those can be added once that issue's closed. In the meanwhile, this gives a pretty significant increase in coverage, and should hopefully avoid breaking the string stdlib by accident.

### CODE REVIEW

File: tests/integration/pass/strings.ncl [hunk 1]
```diff
@@ -11,6 +11,123 @@ let {check, ..} = import "lib/assert.ncl" in
...
+  (1 | string.Stringable) == 1,
+  ("_" | string.NonEmpty) == "_",
+  # string.join
+  string.join " " ["Hello", "世界"] == "Hello 世界",
+  string.join "anything" [] == "",
+  string.join "a" ["b", "n", "n", ""] == "banana",
+  # string.split
+  string.split "," "1,2,3" == ["1", "2", "3"],
+  string.split "" "123" == ["", "1", "2", "3", ""],
+  string.split "❤" "👩‍❤️‍💋‍👨" == [ "👩‍", "️‍💋‍👨" ],
```
Comment (euank) [inline tests/integration/pass/strings.ncl:40]: I'm not sure if this one is correct or not. It seems arguable either way. Same for the one below this one.

File: tests/integration/pass/strings.ncl [hunk 2]
```diff
@@ -11,6 +11,123 @@ let {check, ..} = import "lib/assert.ncl" in
...
+  # string.trim
+  string.trim "  ひげ  " == "ひげ",
+  string.trim "   " == "",
+  string.trim "" == "",
+  string.trim "​ ​ ​ a ​ ​ ​" == "​ ​ ​ a ​ ​ ​", # zero width spaces aren't whitespace
+  string.trim "   \n   new\nline\n" == "new\nline",
+  string.trim "　   	  " == "",
+  # string.chars
+  string.chars "漢字" == ["漢", "字"],
+  string.chars "👨‍❤️‍💋‍👨" == ["👨", "‍", "❤", "️", "‍", "💋", "‍", "👨"], # arguably wrong, splits the ZWJ combined emoji
```
Comment (euank) [inline tests/integration/pass/strings.ncl:52]: This one's also arguable either way in my opinion.

**Review by matthew-healy — APPROVED** [review]
This is a really great suite of tests, thanks so much for adding them!

For the cases you've called out as being unusual, I think it's still reasonable to have tests for the _current_ behaviour, accompanied with a comment (& ideally an issue) pointing out that this should probably change. :+1: 

If I was nitpicking, I'd maybe say that this could be a separate file - `stdlib_string_tests.ncl` - but I don't think that needs to block the PR from being merged. We'll probably change the structure as we add tests for the rest of the stdlib anyhow.

**Review by yannham — APPROVED** [review]
We've been talking about testing the stdlib in a systematic way not later than yesterday :slightly_smiling_face: including some form of property-based testing.

In the meantime, those tests should prevent the kind of error we had on 0.3 for the string library. Thanks for writing them. I agree with @matthew-healy, let's have the controversial tests be adapted to the current behavior. We'll change them in the future if needed (we will have to).

I agree as well with splitting those tests in a separate file, if possible.

**Pull request merged**